contracts/vesting/src/contract.rs

```diff
@@ -389,14 +388,25 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> Result<QueryResponse, Contr
             deps,
         )?),
         QueryMsg::GetAccount { address } => to_binary(&try_get_account(&address, deps)?),
+        QueryMsg::GetMixnode { address } => to_binary(&try_get_mixnode(&address, deps)?),
+        QueryMsg::GetGateway { address } => to_binary(&try_get_gateway(&address, deps)?),
     };
 
     Ok(query_res?)
 }
 
-pub fn try_get_account(address: &str, deps: Deps) -> Result<Option<Account>, ContractError> {
+pub fn try_get_mixnode(address: &str, deps: Deps) -> Result<Option<PledgeData>, ContractError> {
     let account = account_from_address(address, deps.storage, deps.api)?;
-    Ok(Some(account))
+    account.load_mixnode_pledge(deps.storage)
+}
+
+pub fn try_get_gateway(address: &str, deps: Deps) -> Result<Option<PledgeData>, ContractError> {
+    let account = account_from_address(address, deps.storage, deps.api)?;
+    account.load_gateway_pledge(deps.storage)
+}
+
+pub fn try_get_account(address: &str, deps: Deps) -> Result<Account, ContractError> {
```

Comment:
Shouldn't this return `Option<Account>`? What if given address is not vesting?

Reply:
It used to, the thing is that validate will throw an error if the account does not exist, so I think we can just use that, although it was late yesterday, so I can just put the Option back in :)